tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,264 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""DeepLesion dataset."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow_datasets as tfds
+import tensorflow as tf
+import os
+
+
+# info of deeplesion dataset
+_URL = ("https://www.nih.gov/news-events/news-releases/nih-clinical-center-releases-dataset-32000-ct-images")
+
+_DESCRIPTION = """\
+The DeepLesion dataset contains 32,120 axial computed tomography (CT) slices 
+from 10,594 CT scans (studies) of 4,427 unique patients. There are 1–3 lesions 
+in each image with accompanying bounding  boxes  and  size  measurements,  
+adding  up  to  32,735  lesions  altogether.  The  lesion annotations were 
+mined from NIH’s picture archiving and communication system (PACS).
+"""
+
+_CITATION = """\
+@article{DBLP:journals/corr/abs-1710-01766,
+  author    = {Ke Yan and
+               Xiaosong Wang and
+               Le Lu and
+               Ronald M. Summers},
+  title     = {DeepLesion: Automated Deep Mining, Categorization and Detection of
+               Significant Radiology Image Findings using Large-Scale Clinical Lesion
+               Annotations},
+  journal   = {CoRR},
+  volume    = {abs/1710.01766},
+  year      = {2017},
+  url       = {http://arxiv.org/abs/1710.01766},
+  archivePrefix = {arXiv},
+  eprint    = {1710.01766},
+  timestamp = {Mon, 13 Aug 2018 16:48:13 +0200},
+  biburl    = {https://dblp.org/rec/bib/journals/corr/abs-1710-01766},
+  bibsource = {dblp computer science bibliography, https://dblp.org}
+}
+"""
+
+# link for download data
+_BOX_URL = ("https://nihcc.app.box.com/v/DeepLesion")
+
+
+# location where local resource should be
+_IMAGE_DIR = "Images_png"
+
+
+class Deeplesion(tfds.core.GeneratorBasedBuilder):
+  """DeepLesion dataset builder.
+  """
+
+  VERSION = tfds.core.Version('0.1.0')
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=_DESCRIPTION,
+        features=tfds.features.FeaturesDict({
+            "image/name":tfds.features.Text(),
+            "image":tfds.features.Image(shape=(None, None, 1), dtype=tf.uint16, encoding_format='png'),
```

Comment:
This feature description only contains the annotated slices, not the surrounding slices (which are quite useful to know the 3d context).

Consider making a sequence of images corresponding to the bounding box annotation.

Reply:
> This feature description only contains the annotated slices, not the surrounding slices (which are quite useful to know the 3d context).
> 
> Consider making a sequence of images corresponding to the bounding box annotation.

That's a good idea. I will provide this feature next.

Reply:
> This feature description only contains the annotated slices, not the surrounding slices (which are quite useful to know the 3d context).
> 
> Consider making a sequence of images corresponding to the bounding box annotation.

Hi Ouwen, I finished the feature to provide the 3d context. Should I commit it here (to replace this one), or should I provide the 3d context feature using multiply configs or versioning? I'm a little confused about the usage of multiple configs or versioning.

Reply:
Also, I create a new pull request #1342 hoping to clean the commit history and keep updated to the newest official master branch. Let me know should I close that one.

Reply:
You can see the code here 128f0cf7c9ae0a31dac5f0c73212546a40a7d390